src/server/page_server.hpp

```diff
@@ -46,11 +46,13 @@ class page_server
     ~page_server ();
 
     void set_active_tran_server_connection (cubcomm::channel &&chn);
-    void start ();
-    void disconnect_active_tran_server ();
+    void set_passive_tran_server_connection (cubcomm::channel &&chn);
+    void disconnect_tran_server (void *conn);
+    void disconnect_all_tran_server ();
     bool is_active_tran_server_connected () const;
+    bool is_passive_tran_server_connected () const;
     void push_request_to_active_tran_server (page_to_tran_request reqid, std::string &&payload);
-
+    cublog::async_page_fetcher *get_page_fetcher ();
```

Comment:
There is always a page fetcher and returned pointer can never be null. Return a reference instead.
```suggestion
    cublog::async_page_fetcher &get_page_fetcher ();
```